### PR TITLE
Upgrade to Go 1.18

### PR DESCRIPTION
**Pull request by epk**
Signed-off-by: Aditya Sharma <git@adi.run>

### Description of your changes

- Upgrades Go to 1.18
- Fixes some minor lint errors

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cmd/crank/install.go [hunk 1]
```diff
@@ -265,7 +265,7 @@ func (c *installProviderCmd) Run(k *kong.Context, logger logging.Logger) error {
 }
 
 func warnIfNotFound(err error) error {
-	serr, ok := err.(*apierrors.StatusError)
+	serr, ok := err.(*apierrors.StatusError) //nolint:errorlint // we need to be able to extract the underlying typed error
```
Comment (muvaf) [inline cmd/crank/install.go:268]: Not sure why we need the `nolint` here. Could you elaborate?

File: design/design-doc-external-secret-stores.md [hunk 2]
```diff
@@ -264,10 +264,10 @@ metadata:
   name: vault-default
 spec:
   type: Vault
-  # defaultScope used for scoping secrets for cluster scoped resources. 
-  # For example, secrets for MRs will land under 
+  # defaultScope used for scoping secrets for cluster scoped resources.
```
Comment (muvaf) [inline design/design-doc-external-secret-stores.md:267]: Do we need these unrelated changes? Can we discard them?

File: go.sum [hunk 3]
```diff
@@ -296,189 +221,55 @@ github.com/cncf/udpa/go v0.0.0-20210930031921-04548b0d99d4/go.mod h1:6pvJx4me5XP
...
-github.com/coreos/go-iptables v0.5.0/go.mod h1:/mVI274lEDI2ns62jHCDnCyBF9Iwsmekav8Dbxlm1MU=
-github.com/coreos/go-iptables v0.6.0/go.mod h1:Qe8Bv2Xik5FyTXwgIbLAnv2sWSBmvWdFETJConOQ//Q=
 github.com/coreos/go-oidc v2.1.0+incompatible/go.mod h1:CgnwVTmzoESiwO9qyAFEMiHoZ1nMCKZlZ9V6mm3/LKc=
 github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
 github.com/coreos/go-semver v0.3.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
-github.com/coreos/go-systemd v0.0.0-20161114122254-48702e0da86b/go.mod h1:F5haX7vjVVG0kc13fIWeqUViNPyEJxv/OmvnBo0Yme4=
 github.com/coreos/go-systemd v0.0.0-20180511133405-39ca1b05acc7/go.mod h1:F5haX7vjVVG0kc13fIWeqUViNPyEJxv/OmvnBo0Yme4=
 github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e/go.mod h1:F5haX7vjVVG0kc13fIWeqUViNPyEJxv/OmvnBo0Yme4=
 github.com/coreos/go-systemd v0.0.0-20190620071333-e64a0ec8b42a/go.mod h1:F5haX7vjVVG0kc13fIWeqUViNPyEJxv/OmvnBo0Yme4=
-github.com/coreos/go-systemd/v22 v22.0.0/go.mod h1:xO0FLkIi5MaZafQlIrOotqXZ90ih+1atmu1JpKERPPk=
```
Comment (muvaf) [inline go.sum:419]: Do you know why so many removed in this file?

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,6 @@
 module github.com/crossplane/crossplane
 
-go 1.17
+go 1.18
 
 require (
```
Comment (muvaf) [inline go.mod:5]: We can update `crossplane-runtime` version, too. It'd be great if you can run Crossplane with that change and see if it crashes or has anything unexpected in the logs 🙂

File: cmd/crank/install.go (hunk 1 above)
Comment (epk) [reply cmd/crank/install.go:268]: It was a lint error asking to use `errors.As` but that only returns a bool and we'd still have to cast it to `*apierrors.StatusError` 
```
cmd/crank/install.go:268:14: type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors (errorlint)
	serr, ok := err.(*apierrors.StatusError)
```

File: go.sum (hunk 3 above)
Comment (epk) [reply go.sum:419]: Go 1.18 has [module graph pruning](https://go.dev/ref/mod#graph-pruning):
> If the main module is at go 1.17 or higher, the [module graph](https://go.dev/ref/mod#glos-module-graph) used for [minimal version selection](https://go.dev/ref/mod#minimal-version-selection) includes only the immediate requirements for each module dependency that specifies go 1.17 or higher in its own go.mod file, unless that version of the module is also (transitively) required by some other dependency at go 1.16 or below. (The transitive dependencies of go 1.17 dependencies are pruned out of the module graph.)

Comment (muvaf) [reply go.sum:419]: Got it, thanks for the explanation!

**Review by muvaf — APPROVED** [review]
Thanks @epk!
